test/e2e/multihoming.go

```diff
@@ -899,6 +899,85 @@ var _ = Describe("Multi Homing", func() {
 					)
 				})
 			})
+
+			Context("with a trunked configuration", func() {
+				const vlanID = 20
+				BeforeEach(func() {
+					nodes = ovsPods(cs)
+					Expect(nodes).NotTo(BeEmpty())
+
+					// we are setting up the bridge in trunked mode by not
+					// specifying a particular VLAN ID on the network conf
+					netConfig = newNetworkAttachmentConfig(
+						networkAttachmentConfigParams{
+							name:         secondaryNetworkName,
+							namespace:    f.Namespace.Name,
+							topology:     "localnet",
+							cidr:         secondaryLocalnetNetworkCIDR,
+							excludeCIDRs: []string{underlayServiceIP + "/32"},
+						})
+
+					By("setting up the localnet underlay with a trunked configuration")
+					Expect(setupUnderlay(nodes, secondaryInterfaceName, netConfig)).To(Succeed(), "configuring the OVS bridge")
+
+					By(fmt.Sprintf("creating a VLAN interface on top of the bridge connecting the cluster nodes with IP: %s", underlayIP))
+					cli, err := client.NewClientWithOpts(client.FromEnv)
+					Expect(err).NotTo(HaveOccurred())
+
+					gatewayIP, err := getNetworkGateway(cli, dockerNetworkName)
+					Expect(err).NotTo(HaveOccurred())
+
+					underlayBridgeName, err = findInterfaceByIP(gatewayIP)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(createVLANInterface(underlayBridgeName, strconv.Itoa(vlanID), &underlayIP)).To(
+						Succeed(),
+						"create a VLAN interface on the bridge interconnecting the cluster nodes",
+					)
+
+					By("starting a service, connected to the underlay")
+					cmdWebServer = exec.Command("python3", "-m", "http.server", "--bind", underlayServiceIP, strconv.Itoa(port))
```

Comment:
it doesn't introduce; the dependency is already there: https://github.com/ovn-org/ovn-kubernetes/blob/08a965003b2b3da204278a7161850167c0c559d0/test/e2e/multihoming.go#L697

I'm just extending it.

I agree it would be a good follow up to remove these dependencies altogether and use a golang http server.